File: lab12_texteditor/TextEditor/TextEditor/EditorMemento.cs
namespace TextEditor;

public class EditorMemento
{
    public readonly string _text;
    public readonly int _cursorPosition;
    public readonly string _selection;
 
    public EditorMemento(string text, int cursorPosition, string selection)
    {
        _text = text;
        _cursorPosition = cursorPosition;
        _selection = selection;
    }
     
    public string GetText()
    {
        return _text;
    }
     
    public int GetCursorPosition()
    {
        return _cursorPosition;
    }
     
    public string GetSelection()
    {
        return _selection;
    }
}

File: Text Editor/src/texteditor/EditorMemento.java
package texteditor;

import java.util.List;

public class EditorMemento {
    private final String content;
    private final int cursorPosition;
    private final List<String> selections;

    // Implementing the Constructor
    public EditorMemento(String content, int cursorPosition, List<String> selections) {
        this.content = content;
        this.cursorPosition = cursorPosition;
        this.selections = selections;
    }

    // Getters to retrieve stored state
    public String getContent() {
        return content;
    }

    public int getCursorPosition() {
        return cursorPosition;
    }

    public List<String> getSelections() {
        return selections;
    }
}
